### PR TITLE
Resolve: update to new publisher

### PR DESCRIPTION
**Pull request by jakubjezek001**
> [!NOTE] 
> Description is work in progress

## Changelog Description
Resolve with adapted latest publisher tool.

## Additional info
Adaptation of https://github.com/ynput/OpenPype/pull/5709 (thanks @bigroy for the kick-off!)

# TODO:

- [ ] - @BigRoy reported that there is auto timeline created which should not 
- [ ] - @BigRoy suggested warning message if there is not anything to be used for creation
- [ ] - @BigRoy suggested to add plugin info text under `?`
- [ ] - Spliting product types into:
        	- **primary**: `shot` - instance with hierarchy, future context data and attributes placeholder
            - **secondary**: `plate`, `audio`, `review` - instance with media
- [ ] - Hierarchy and context data should be linked from Shot instance for all secondary product types Traypublisher (TP)
- [ ] - Each secondary product type which will be created from a clip should have own creator class - like in TP
- [x] - OTIO should be using native format rather then our implementation - this way we will be ready for Tony's (@tweak-wtf) retiming.

## Testing notes:
To be created

### CODE REVIEW

**Comment by BigRoy** [comment]
Nice!

1. Just opening the Publisher creates an empty timeline (easiest to test in empty project) - it should not do anything.
2. When Clicking Create for Create Publishable Clip but there's no valid clips in timeline nothing happens - and there's no warning or printed message in console. Might be good to provide a Creation Error if it ends up doing nothing. Easily reproducable in empty timeline or timeline without any Chocolate color clips if that toggle is enabled.
3. Still seems to suffer from the fact that it tries to set an invalid folder path. Which I assume is similar to where the [original PR](https://github.com/ynput/OpenPype/pull/5709) left things unfinished.
image
4. Deleting the instances gives me this error:
image
5. Might be nice to add a "detailed description" for this creator - for the question mark top right in publisher.

File: client/ayon_core/hosts/resolve/api/lib.py [hunk 1]
```diff
@@ -899,13 +928,53 @@ def get_otio_clip_instance_data(otio_timeline, timeline_item_data):
...
+        anatomy (ayon_core.pipeline.Anatomy)[optional]: Anatomy object
+        timeline (resolve.Timeline)[optional]: resolve's object
+
+    Returns:
+        str: temporary otio filepath
+    """
+    from . import bmdvr
+    resolve_project = get_current_resolve_project()
+    timeline = resolve_project.GetCurrentTimeline()
+    timeline_name = timeline.GetName()
```
Comment (BigRoy) [inline client/ayon_core/hosts/resolve/api/lib.py:952]: This should be using the `timeline` that's passed along as an argument if not `None`.
```suggestion
    if timeline is None:
        timeline = resolve_project.GetCurrentTimeline()
        if not timeline:
            raise RuntimeError("No current timeline")
    timeline_name = timeline.GetName()
```

File: server_addon/resolve/client/ayon_resolve/RESOLVE_API_v19.0B-build20.txt [hunk 2]
```diff
@@ -102,8 +106,11 @@ Resolve
   ExportRenderPreset(presetName, exportPath)      --> Bool               # Export a preset to a given path (string) if presetName(string) exists.
   ImportBurnInPreset(presetPath)                  --> Bool               # Import a data burn in preset from a given presetPath (string)
   ExportBurnInPreset(presetName, exportPath)      --> Bool               # Export a data burn in preset to a given path (string) if presetName (string) exists.
+<<<<<<<< HEAD:server_addon/resolve/client/ayon_resolve/RESOLVE_API_v18.6.5-build7.txt
+========
```
Comment (BigRoy) [inline server_addon/resolve/client/ayon_resolve/RESOLVE_API_v19.0B-build20.txt:110]: This looks like some merge conflicts gone wrong?

File: client/ayon_core/hosts/resolve/api/pipeline.py [hunk 3]
```diff
@@ -260,3 +263,138 @@ def on_pyblish_instance_toggled(instance, old_value, new_value):
     # Whether instances should be passthrough based on new value
     timeline_item = instance.data["item"]
     set_publish_attribute(timeline_item, new_value)
+
+
+class HostContext:
```
[the remark this answers is not in the document]
Comment (jakubjezek001) [reply client/ayon_core/hosts/resolve/api/pipeline.py]: I am not sure where should be this comment on. It seems the code had changed a lot.

[the remark this answers is not in the document]
Comment (BigRoy) [reply client/ayon_core/hosts/resolve/api/pipeline.py]: It seems to be on the right spot still. Basically, what is the `HostContext` class and why do we need it? Would be good to have a docstring to describe its usage.

**Comment by iLLiCiTiT** [comment]
PR moved to ayon-resolve repository https://github.com/ynput/ayon-resolve/pull/3

**Pull request closed without merging**